### PR TITLE
[PhpunitBridge] Restore memoization

### PR DESCRIPTION
**Pull request by greg0ire**
| Q             | A
| ------------- | ---
| Branch?       | 4.3
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

I must have inadvertently dropped it in 1c73f9cfedad7fb7eb6ab0b1230d0219685ead0f

### CODE REVIEW

File: src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php [hunk 1]
```diff
@@ -216,7 +216,7 @@ private function getConfiguration()
             return $this->configuration;
         }
         if (false === $mode = $this->mode) {
-            $mode = getenv('SYMFONY_DEPRECATIONS_HELPER');
+            $mode = $this->mode = getenv('SYMFONY_DEPRECATIONS_HELPER');
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php]: are you sure? isn't it memoization taken care of by the "configuration" property now?

Comment (greg0ire) [reply src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php]: Oh you're correct! I'll update the PR to just remove that property then.

**Comment by greg0ire** [comment]
Looks like I broke something :confused:

File: src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php [hunk 2]
```diff
@@ -215,9 +214,7 @@ private function getConfiguration()
         if (null !== $this->configuration) {
             return $this->configuration;
         }
-        if (false === $mode = $this->mode) {
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php]: Not sure this is legit: the property is used L88.

Comment (greg0ire) [reply src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php]: Oh, you're right, good catch! Closing then.

**Pull request closed without merging**